open-hackathon-server/src/hackathon/hack/hackathon_manager.py

```diff
@@ -736,6 +817,54 @@ def __set_basic_property(self, hackathon, prop):
             self.db.add_object(config)
         self.db.commit()
 
+    def __create_hackathon_notice(self, hackathon, notice_event, args=None):
+        """
+        create hackathon notice
+        :type hackathon: Hackathon
+        :param hackathon: an existing Hackathon object that the notice belongs to
+
+        :type notice_event: Class HACK_NOTICE_EVENT
+        :param notice_event: a specfic event that the notice is triggered by
+
+        :return: hackathon_notice in dict
+        """
+        hackathon_notice = HackathonNotice(hackathon_id=hackathon.id, 
+                                           content="",
+                                           link="",
+                                           event=notice_event,
+                                           create_time=self.util.get_now(),
+                                           update_time=self.util.get_now())
+
+        if notice_event == HACK_NOTICE_EVENT.MANUAL:
+            hackathon_notice.content = args.get('content', '')
+            hackathon_notice.link = args.get('link', '')
+        elif notice_event == HACK_NOTICE_EVENT.HACK_CREATE:
+            hackathon_notice.content = u"Hachathon: %s 创建成功" %(hackathon.name)
+        elif notice_event == HACK_NOTICE_EVENT.HACK_EDIT:
+            hackathon_notice.content = u"Hachathon: %s 信息变更" %(hackathon.name)
+        elif notice_event == HACK_NOTICE_EVENT.HACK_ONLINE:
+            hackathon_notice.content = u"Hachathon: %s 正式上线" %(hackathon.name)
+        elif notice_event == HACK_NOTICE_EVENT.HACK_OFFLINE:
+            hackathon_notice.content = u"Hachathon: %s 下线" %(hackathon.name)
+        elif notice_event == HACK_NOTICE_EVENT.EXPR_JOIN and args.get('user_id'):
+            user_id = int(args.get('user_id'))
+            user = self.user_manager.get_user_by_id(user_id)
+            hackathon_notice.content = u"用户 %s 开始编程" %(user.nickname)
+        else:
+            pass
+
+        # every event belongs to a specific type, which influences the icons/descriptions of the notice shown at front-end
+        if notice_event in [HACK_NOTICE_EVENT.EXPR_JOIN]:
+            hackathon_notice.type = HACK_NOTICE_TYPE.EXPERIMENT
+        else:
+            hackathon_notice.type = HACK_NOTICE_TYPE.HACKATHON
```

Comment:
I suggest passing the `type` as parameter. The caller knows the best about `type` and `event`. See comments before